From 2e0c2c91507c46fd375381ca96664ec9a32e5407 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E7=9F=B3=E5=8D=9A=E6=96=87?= <sbw@sbw.so>
Date: Wed, 20 Mar 2019 23:22:58 +0800
Subject: [PATCH] add option for window location

---
 plugin/asyncrun.vim | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/plugin/asyncrun.vim b/plugin/asyncrun.vim
index 6d2ca0a..b805af5 100644
--- a/plugin/asyncrun.vim
+++ b/plugin/asyncrun.vim
@@ -182,6 +182,10 @@ if !exists('g:asyncrun_save')
 	let g:asyncrun_save = 0
 endif
 
+if !exists('g:asyncrun_window_location')
+	let g:asyncrun_window_location = 'botright'
+endif
+
 if !exists('g:asyncrun_stdin')
 	let g:asyncrun_stdin = has('win32') || has('win64') || has('win95')
 endif
@@ -1375,12 +1379,12 @@ function! asyncrun#quickfix_toggle(size, ...)
 		endif
 	elseif l:mode == 1
 		if s:quickfix_open == 0
-			exec 'botright copen '. ((a:size > 0)? a:size : ' ')
+			exec g:asyncrun_window_location 'copen '. ((a:size > 0)? a:size : ' ')
 			wincmd k
 		endif
 	elseif l:mode == 2
 		if s:quickfix_open == 0
-			exec 'botright copen '. ((a:size > 0)? a:size : ' ')
+			exec g:asyncrun_window_location 'copen '. ((a:size > 0)? a:size : ' ')
 			wincmd k
 		else
 			silent! cclose
-- 
2.21.0

